src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -142,22 +156,69 @@ class TransportCcEngine(
         }
     }
 
+    /** Trim old sent packet details from the map.  Caller should be synchronized on [sentPacketsSyncRoot]. */
+    private fun cleanupSentPacketDetails(newSeq: Int): Boolean {
+        if (sentPacketDetails.isEmpty()) {
+            return true
+        }
+
+        val latestSeq = sentPacketDetails.lastKey()
+
+        /* If our sequence numbers have jumped by a quarter of the number space, reset the map. */
+        val seqJump = getSequenceNumberDelta(newSeq, latestSeq)
+        if (seqJump >= 0x4000 || seqJump <= -0x4000) {
+            sentPacketDetails.clear()
+            return true
+        }
+
+        val threshold = applySequenceNumberDelta(latestSeq, -MAX_OUTGOING_PACKETS_HISTORY)
+
+        if (isOlderSequenceNumberThan(newSeq, threshold)) {
+            return false
+        }
+
+        val it = sentPacketDetails.navigableKeySet().iterator()
```

Comment:
This might be a good use case for `with`:
```
with(sentPacketDetails.navigableKeySet().iterator()) {
  while (hasNext()) {
    val key = next()
    ...
  }
}
```

Reply:
Ok.